File: python/src/tiny_claw/tools/registry.py
from abc import ABC, abstractmethod

from tiny_claw.schema import ToolCall, ToolDefinition, ToolResult
from tiny_claw.tools.base import BaseTool


class Registry(ABC):
    """对照 go/internal/tools/registry.go"""

    @abstractmethod
    def register(self, tool: BaseTool) -> None:
        raise NotImplementedError

    @abstractmethod
    def get_available_tools(self) -> list[ToolDefinition]:
        raise NotImplementedError

    @abstractmethod
    def execute(self, call: ToolCall) -> ToolResult:
        raise NotImplementedError


class RegistryImpl(Registry):
    def register(self, tool: BaseTool) -> None:
        raise NotImplementedError("TODO: 对照 go/internal/tools/registry.go 实现")

    def get_available_tools(self) -> list[ToolDefinition]:
        raise NotImplementedError("TODO: 对照 go/internal/tools/registry.go 实现")

    def execute(self, call: ToolCall) -> ToolResult:
        raise NotImplementedError("TODO: 对照 go/internal/tools/registry.go 实现")

File: ts/src/tools/registry.ts
import type { ToolCall, ToolDefinition, ToolResult } from "../schema/message";
import type { BaseTool } from "./base-tool";

/** 对照 go/internal/tools/registry.go */
export interface Registry {
  register(tool: BaseTool): void;
  getAvailableTools(): ToolDefinition[];
  execute(call: ToolCall): Promise<ToolResult>;
}

export class RegistryImpl implements Registry {
  register(_tool: BaseTool): void {
    throw new Error("TODO: 对照 go/internal/tools/registry.go 实现");
  }

  getAvailableTools(): ToolDefinition[] {
    throw new Error("TODO: 对照 go/internal/tools/registry.go 实现");
  }

  execute(_call: ToolCall): Promise<ToolResult> {
    throw new Error("TODO: 对照 go/internal/tools/registry.go 实现");
  }
}
